src/core/SampleBuffer.cpp

```diff
@@ -131,6 +131,79 @@ SampleBuffer::SampleBuffer( const f_cnt_t _frames )
 
 
 
+SampleBuffer::SampleBuffer(SampleBuffer& orig):
+	m_audioFile(orig.m_audioFile),
+	m_origData(MM_ALLOC(sampleFrame, orig.m_origFrames)),
+	m_origFrames(orig.m_origFrames),
+	m_data(MM_ALLOC(sampleFrame, orig.m_frames)),
+	m_frames(orig.m_frames),
+	m_startFrame(orig.m_startFrame),
+	m_endFrame(orig.m_endFrame),
+	m_loopStartFrame(orig.m_loopStartFrame),
+	m_loopEndFrame(orig.m_loopEndFrame),
+	m_amplification(orig.m_amplification),
+	m_reversed(orig.m_reversed),
+	m_frequency(orig.m_frequency),
+	m_sampleRate(orig.m_sampleRate)
+{
+	orig.m_varLock.lockForRead();
+	//Deep copy m_origData and m_data from original
+	memcpy(m_origData, orig.m_origData, m_origFrames* BYTES_PER_FRAME);
+	memcpy(m_data, orig.m_data, m_frames* BYTES_PER_FRAME);
+	orig.m_varLock.unlock();
+}
```

Comment:
IIRC, doesn't the lock only protect m_data? And the argument's m_data is neither read nor written in the initializers.

Reply:
I thought it was protecting the whole object, since other variables are related to `m_data` as well (like frame counts), but I'm not 100% sure to be honest, will try to investigate further!

Reply:
From my check it doesn't protect anything else currently, but that doesn't necessarily mean it shouldn't. I think it's fine to leave this as is though, unless anyone disagrees?

Reply:
Sorry I couldn't look into it further, this week I haven't had much spare time. If it doesn't protect all the other members, then I think it's fine to leave it as is and change it when we change `SampleBuffer` to protect them (if we need to)

Reply:
Unresolving because Ian stated on Discord that
> the lock actually protects members like m_frames. It's just that they depend on m_data and are only changed when it is changed as well

Will fix this and the style issues M374LX brought up in the next commit.